### ISSUE TITLE
Building with -debug panics.

### ISSUE THREAD

**Issue by JesseRMeyer** [odin-lang/Odin#1260]
```
F:\Odin>odin build examples/demo/demo.odin -debug
src/llvm_backend_debug.cpp(290): Panic: Type_Named should be handled in lb_debug_type separately
```

odin report on master branch:
```
Odin: dev-2021-11
        OS:   Windows 10 Unknown Edition (00000065) (version: 21H1), build 19043.1320
        CPU:  Intel(R) Core(TM) i5-3570K CPU @ 3.40GHz
        RAM:  32717 MiB
```

**Comment by Kelimion** [comment]
Confirmed:

W:\Odin>odin build examples\demo -debug
W:\Odin\src\llvm_backend_debug.cpp(290): Panic: Type_Named should be handled in lb_debug_type separately

Also added Windows 10 Home Core (0x65) detection to `odin report` just now.

**Comment by Kelimion** [comment]
Narrowed it down to two culprits:

Commenting out `matrix_type()` no longer hits the aforementioned assert. Instead, we now get a new error: `Kernel32.lib(KERNEL32.dll) : fatal error LNK1103: debugging information corrupt; recompile module`

That one is triggered by `soa_struct_layout()`.

Comment both out and the demo will compile with debug information.

**Comment by Kelimion** [comment]
`llvm_backend_debug.cpp` assert has been fixed. Link error from `soa_struct_layout()` still happens, as confirmed by CI.

**Comment by Kelimion** [comment]
Narrowed down this linker bug. If I comment out the soa related processing in llvm_backend_debug.cpp, the demo will run with debug info and soa_struct_styles() compiled in without that link error.

I figure this means that the debug info will not be correct for SOA types, but it means that something here is what breaks the debug info to begin with.

![image](https://user-images.githubusercontent.com/194614/139723402-bed9fd39-5bb1-4f9c-8ce2-48b09e380985.png)

**Comment by Kelimion** [comment]
Fixed in b01e0fbbc209110cf49b980149edb73ece475af3.
